tests/integration/endpoint_test.go

```diff
@@ -14,17 +14,33 @@ func endpointTests(t *testing.T, when spec.G, it spec.S) {
 	var service testutil.TestService
 	var stagedService testutil.TestService
 
+	it.Before(func() {
+		service.Create(t, "ibuildthecloud/demo:v1")
+		stagedService = service.Stage("ibuildthecloud/demo:v3", "v3")
+	})
 	it.After(func() {
 		service.Remove()
 		stagedService.Remove()
 	})
 
 	when("a service is running and another is staged", func() {
 		it("should have endpoints that are available on both", func() {
-			service.Create(t, "ibuildthecloud/demo:v1")
-			stagedService = service.Stage("ibuildthecloud/demo:v3", "v3")
 			assert.Equal(t, "Hello World", service.GetEndpoint())
 			assert.Equal(t, "Hello World v3", stagedService.GetEndpoint())
+			kubeUrlv0 := service.GetKubeEndpointUrlByRevision("v0")
+			kubeUrlv3 := stagedService.GetKubeEndpointUrlByRevision("v3")
+			rioUrlv0 := service.GetEndpointUrl()
+			rioUrlv3 := stagedService.GetEndpointUrl()
+			assert.Equal(t, kubeUrlv0, rioUrlv0)
+			assert.Equal(t, kubeUrlv3, rioUrlv3)
+
+			assert.Equal(t, service.GetEndpointResponse(kubeUrlv0), service.GetEndpointResponse(rioUrlv0))
+		})
+		it("should have the global endpoint properly created", func(){
```

Comment:
Lets move stagedService above so it doesn't get created here, only in first test. Might break this one actually if it does, as weight can move over to staged one and change endpoint response.